### DISCUSSION TITLE
Strange Min-max cell behaviour

### DISCUSSION THREAD

**Discussion by hansgeorgk** [Louisvdw/dbus-serialbattery#451] [Q&A]
Hi,

while charging, I see a big difference between Min/Max voltage and when Max voltage is getting close to 3.5V, CCCM_CV_ENABLE is reducing CCL to close to "0". Even so the Cells are far from being full, charging is basically stopped :-(
![Bildschirm­foto 2023-02-05 um 07 29 19](https://user-images.githubusercontent.com/100286224/216979613-cb61f45e-c7d5-4f63-b6c7-b779cee6bbf7.png)
![Bildschirm­foto 2023-02-05 um 07 29 28](https://user-images.githubusercontent.com/100286224/216979630-05a499ba-7c6f-40a3-899e-295b1a68a590.png)

I see similar behaviour when discharging at high load, min voltage of a single cell is dropping to MIN_CELL_VOLTAGE and system is stopped, even though total voltage is still above 25V

![Bildschirm­foto 2023-01-17 um 07 00 00](https://user-images.githubusercontent.com/100286224/216979779-e15c8fd7-3389-472a-9403-725a6d938d1b.png)

I suspect this is one cell going crazy and I would like to verify this.
Is there a way to get the historical data of individual cell voltages?

cheers
Hans-Georg

**Comment by najaB76** [comment]
Possibly related, but I've noticed that the max cell voltage is improbably high when the SOC gets above 90%.  From dbus-spy:

Voltages/Cell1         3.512
Voltages/Cell2        36.276
Voltages/Cell3        36.281
Voltages/Cell4        36.277
Voltages/Cell5         3.493
Voltages/Cell6         3.509
Voltages/Cell7         3.513
Voltages/Cell8        36.278

This is with the EG4 LifePower battery if that helps.

Here are the voltages as read from the battery directly:

![image](https://user-images.githubusercontent.com/96979341/218837215-7b3cef70-6de9-4d7c-ae01-b6ed68a87f42.png)
